packages/common/cumulus-api-client/CumulusApiClient.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const got = require('got');
+const normalizeUrl = require('normalize-url');
+const { decryptBase64String, encrypt } = require('@cumulus/aws-client/KMS');
+const { dynamodbDocClient } = require('@cumulus/aws-client/services');
+const { decode } = require('jsonwebtoken');
+const Logger = require('@cumulus/logger');
+const CumulusApiClientError = require('./CumulusApiClientError');
+const CumulusAuthTokenError = require('./CumulusAuthTokenError');
+
+const logger = new Logger({});
+
+class CumulusApiClient {
+  constructor(config = {}, requiredKeys = ['kmsId', 'baseUrl', 'authTokenTable', 'tokenSecretName']) {
+    if (!requiredKeys.every((key) => Object.keys(config).includes(key))) {
+      throw new CumulusApiClientError(
+        `Invalid config ${JSON.stringify(config)} - required keys are ${JSON.stringify(requiredKeys)}`
+      );
+    }
+    const defaultConfig = {
+      tokenExpireBuffer: 5 * 60,
+      tokenSecretName: 'cachedBearerTokenDefault'
+    };
+    this.config = { ...defaultConfig, ...config };
+    this.config.baseUrl = normalizeUrl(config.baseUrl);
+    this.Error = CumulusApiClientError;
+  }
+
+  /**
+   * Calls the cumulus api (with 'get') with optional token authentication retries
+   * @param {string} url - Cumulus API endpoint to call
+   * @param {integer} authRetry - number of times to retry on auth expiry failure.
+   *                              Should be set to 1 for launchpad oauth to account
+   *                              for ouath expiration failures
+   * @returns {Promise<Object>} - Returns response object from got.get
+   */
+  async get(url, authRetry = 1) {
+    const headers = { Authorization: `Bearer ${await this.getCacheAuthToken()}` };
+    try {
+      const gotReturn = await got.get(`${this.config.baseUrl}/${url}`, { headers });
+      return gotReturn;
+    } catch (error) {
+      if (authRetry !== 0 && error.message === 'Access token has expired') {
+        logger.info('API Client access token expired, generating new token');
+        await this.getCacheAuthToken();
+        return this.get(url, authRetry - 1);
+      }
+      throw error;
+    }
+  }
+
+  async createNewAuthToken() {
+    throw new CumulusApiClientError('Not implemented in the base class');
+  }
+
+  /**
+   * gets an auth token back from the dynamo cache table
+   *
+   * @returns {Promise<Object>} - decrypted auth token from the auth token table
+   */
+  async _getAuthTokenRecord() {
+    const params = {
+      TableName: this.config.authTokenTable,
+      Key: {
+        tokenAlias: this.config.tokenSecretName
+      }
+    };
+    const tokenResponse = await dynamodbDocClient().get(params).promise();
+    if (!tokenResponse.Item) {
+      throw new CumulusAuthTokenError(`No bearer token with alias '${this.config.tokenSecretName}'
+      found in ${this.config.authTokenTable}`);
+    }
+    return decryptBase64String(tokenResponse.Item.bearerToken);
```

Comment:
Realized due to CI failure, we need to add a catch here to throw a CumulusAuthTokenError  if decryption fails, as deployment can result in a changed/new key.

Reply:
Right - in which case all cached keys are invalidated and it throws a KMS error, which will never result in a new key.   👎